python_modules/dagster/dagster/_core/execution/run_metrics_thread.py

```diff
@@ -63,41 +66,48 @@ def _get_container_metrics(
     cpu_usage = metrics.get("cpu_usage")
     cpu_usage_ms = None
     if cpu_usage is not None:
-        cpu_usage_ms = cpu_usage * 1000  # convert from seconds to milliseconds
+        cpu_usage_ms = check.not_none(cpu_usage) * 1000  # convert from seconds to milliseconds
 
     cpu_limit_ms = None
     if cpu_quota_us and cpu_quota_us > 0 and cpu_period_us and cpu_period_us > 0:
         # Why the 1000 factor is a bit counterintuitive:
         # quota / period -> fraction of cpu per unit of time
         # 1000 * quota / period -> ms/sec of cpu
-        cpu_limit_ms = (1000.0 * cpu_quota_us) / cpu_period_us
+        cpu_limit_ms = (1000.0 * check.not_none(cpu_quota_us)) / check.not_none(cpu_period_us)
 
     cpu_usage_rate_ms = None
     measurement_timestamp = metrics.get("measurement_timestamp")
-    if (
-        previous_cpu_usage_ms
-        and cpu_usage_ms
-        and previous_measurement_timestamp
+    if (previous_cpu_usage_ms and cpu_usage_ms and cpu_usage_ms > previous_cpu_usage_ms) and (
```

Comment:
i'm having trouble following the "cpu_usage_ms > previous_cpu_usage_ms" part - does this mean we are only tracking the rate if it is positive?

Reply:
@gibsondan - yes, the underlying cgroup construct for the cpu is a counter. 
you need to take the derivative to get cpu usage at a given point. however, i tried using the datadog derivative function but it behaves funny. so I am computing the rate myself (same thing as Chris was doing), and forwarding the computation to datadog. the underlying metric doesn't have a lot of value otherwise, really, so not sending it.

Reply:
actually mis-read your comment, and changed the comparison operator to >=; I don't think it can practically happen, given that the same code that's capturing this information will be consuming cpu cycles too. but it definitely makes more sense.